src/components/instance-form-builder/InstanceFormBuilder.jsx

```diff
@@ -21,11 +21,12 @@ class InstanceFormBuilder extends React.Component {
         super(props);
 
         const {
+            match,
             location,
             appConfig: { encryptionKey },
         } = props;
 
-        const isEdit = location.instance !== undefined;
+        const isEdit = match.params.formFor === "edit";
```

Comment:
Property `isEdit` depends solely on `props`, strictly speaking, it should not be in the state, maybe a class constant?